python/dgl/distributed/partition.py

```diff
@@ -42,6 +44,49 @@ def _get_part_ranges(id_ranges):
         res[key] = np.concatenate([np.array(l) for l in id_ranges[key]]).reshape(-1, 2)
     return res
 
+
+def _convert_str_to_tuple_key(part_d):
+    copy_d = copy.deepcopy(part_d)
+    # Iterate over d, modify on copy_d
+
+    def _recursive_convert_str_to_tuple_key(part_d, key_tuple=tuple()):
+        if isinstance(part_d, dict):
+            for k, v in part_d.items():
+                sub_d = copy_d
+                for key in key_tuple:
+                    sub_d = sub_d[key]
+                if k.startswith("tuple"):
+                    try:
+                        tuple_key = literal_eval(k[5:])
+                        sub_d[tuple_key] = sub_d[k]
+                        sub_d.pop(k)
+                    except ValueError:
+                        print("Error parse: {}".format(k))
+                _recursive_convert_str_to_tuple_key(v, key_tuple+(k,))
+        # return copy_d
+    _recursive_convert_str_to_tuple_key(part_d)
+    return copy_d
+
+
+def _convert_tuple_key_to_str(part_d):
```

Comment:
Please add a comment (or docstring?) that the edge type keys should always be in canonical form.